zomes/chat/src/lib.rs

```diff
@@ -123,6 +178,41 @@ fn list_messages(list_messages_input: ListMessagesInput) -> ExternResult<ListMes
     Ok(message::handlers::list_messages(list_messages_input)?)
 }
 
+#[derive(Debug, Serialize, Deserialize, SerializedBytes, Clone)]
+#[serde(rename_all = "camelCase")]
+pub struct ChannelMessages {
+    pub channel: ChannelData,
+    pub messages: Vec<MessageData>,
+}
+
+#[derive(Debug, Serialize, Deserialize, SerializedBytes, Clone)]
+#[serde(rename_all = "camelCase")]
+pub struct AllMessagesList {
```

Comment:
you could just have this as `pub struct AllMessagesList(Vec<ChannelMessages>)`

Reply:
good idea, fixed